.github/workflows/build_and_test.yml

```diff
@@ -552,7 +552,9 @@ jobs:
         python3.9 -m pip install 'pandas-stubs==1.2.0.53'
     - name: Install R linter dependencies and SparkR
       run: |
-        apt-get install -y libcurl4-openssl-dev libgit2-dev libssl-dev libxml2-dev
+        apt-get install -y libcurl4-openssl-dev libgit2-dev libssl-dev libxml2-dev \
+          libfontconfig1-dev libharfbuzz-dev libfribidi-dev libfreetype6-dev libpng-dev \
```

Comment:
Unfortunately, it seems that we are hitting another issue.
```
Get:47 http://archive.ubuntu.com/ubuntu focal/universe amd64 libgit2-dev amd64 0.28.4+dfsg.1-2 [691 kB]
[98](https://github.com/apache/spark/runs/7455057445?check_suite_focus=true#step:10:99)
E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/f/freetype/libfreetype-dev_2.10.1-2ubuntu0.1_amd64.deb  404  Not Found [IP: 185.125.190.36 80]
[99](https://github.com/apache/spark/runs/7455057445?check_suite_focus=true#step:10:100)
E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/f/freetype/libfreetype6-dev_2.10.1-2ubuntu0.1_amd64.deb  404  Not Found [IP: 185.125.190.36 80]
```

Reply:
Currently, it's broken in all branches `master/3.3/3.2/3.1`.

Reply:
Reverted from master/3.3/3.2/3.1

@zhengruifeng mind taking a look please?

Reply:
Try to add the `apt update` before R lint deps installation.

This issue might be 3 reasons.

1. Remote ubuntu upstream source is unstable.
2. Remote ubuntu source list upgrade but local source list is outdate.

If reason is 2, we can add apt update to solve.

Reply:
is it related to network issue? or ubuntu repository?

Reply:
https://github.com/Yikun/spark/pull/135

See this, I submit a pr from my phone, looks like apt install work as expected.